### ISSUE TITLE
panic: reflect.Set: value of type interface {} is not assignable to type []string

### ISSUE THREAD

**Issue by bep** [goccy/go-yaml#601]
I have ported the TOML config in [hugoreleaser](https://github.com/gohugoio/hugoreleaser) to YAML using this YAML library. The primary motivation behind this was to be able to use YAML anchors/aliases get Hugo's release config a little more manageable.

Unfortunately my setup crashes:

See https://go.dev/play/p/lNf3br5S0sF

```go
package main

import (
	"strings"

	yaml "github.com/goccy/go-yaml"
)

type Build struct {
	Definitions   map[string]any `json:"definitions"`
	BuildSettings BuildSettings  `json:"build_settings"`
}

type BuildSettings struct {
	Binary  string   `json:"binary"`
	Flags   []string `json:"flags"`
	Env     []string `json:"env"`
	Ldflags string   `json:"ldflags"`
}

func main() {
	config := `
definitions:
  myenv: &my_env
    - VAR1=1
    - VAR2=2
build_settings:
  binary: hugoreleaser
  flags:
    - -buildmode
    - exe
  env: *my_env
  ldflags: "-s -w -X main.tag=${HUGORELEASER_TAG}"

`
	d := yaml.NewDecoder(strings.NewReader(config),
		yaml.DisallowUnknownField(),
	)

	var cfg Build

	err := d.Decode(&cfg)
	if err != nil {
		panic(err)
	}
}

```

**Comment by bep** [comment]
Some added notes:

This works:

```go
package main

import (
	"fmt"
	"strings"

	yaml "github.com/goccy/go-yaml"
)

type Build struct {
	Definitions   map[string]any `json:"-"`
	BuildSettings BuildSettings  `json:"build_settings"`
}

type BuildSettings struct {
	Binary  string   `json:"binary"`
	Flags   []string `json:"flags"`
	Env     []any    `json:"env"`
	Ldflags string   `json:"ldflags"`
}

func main() {
	config := `
definitions:
  myenv: &my_env
    - VAR1=1
    - VAR2=2
build_settings:
  binary: hugoreleaser
  flags:
    - -buildmode
    - exe
  env: *my_env
  ldflags: "-s -w -X main.tag=${HUGORELEASER_TAG}"
`
	d := yaml.NewDecoder(strings.NewReader(config))

	var cfg Build

	err := d.Decode(&cfg)
	if err != nil {
		panic(err)
	}
	fmt.Println("Env:", cfg.BuildSettings.Env)
}
```

But that prevents me from using `yaml.DisallowUnknownField()`.

**Comment by goccy** [comment]
@bep Thank you for your reports !!! I've fixed this problem with #602

**Comment by goccy** [comment]
This problem has been fixed in the latest release ( v1.15.13 ) ! Thank you 😄
